clang/include/clang/Basic/Attr.td

```diff
@@ -1782,7 +1806,8 @@ def SYCLAddIRAttrCommonMembers : SYCLAddIRAttrMemberCodeHolder<[{
       NameLValue = NameCE->getAPValueResult();
 
     if (!NameLValue.isLValue())
-      return std::nullopt;
+      return getValidAttributeNameAsString(NameLValue, Context,
```

Comment:
Note for reviewers: this particular change was done to avoid assertion failure due to invalid attribute name (`nullopt` being returned) on the following test case:

https://github.com/intel/llvm/blob/c02a98cc705ce056268789382ff9f9afe74c747e/clang/test/SemaSYCL/attr-add-ir-attributes.cpp#L78

Reply:
I'm a little confused. Why did your change cause this and the fail mentioned below to break? I understand `getFilteredAttributeNameValuePairs` is now called earlier but I don't understand why that causes these specific issues.

Reply:
> I'm a little confused. Why did your change cause this and the test mentioned below to break?

I honestly don't understand that myself. I can do a bit more of a debugging here to see why we see this particular AST and not another. My current guess is that it can be related to evaluation of `constexpr` variables, i.e. by the time we parse this attribute, we haven't done some processing of `CEAttrName1` (due to lazy nature of that processing), but it is always done before we reach CodeGen and therefore we haven't seen this problem before

Reply:
Ok, I seem to have figured it out. The issue is reproducible even without my changes, it is just hidden by the test without my changes. If you try to compile the following file with plain `intel/llvm`:
```
$ cat p.cpp
constexpr const char CEAttrName1[] = "CEAttr1";

__attribute__((sycl_device)) [[__sycl_detail__::add_ir_attributes_function(CEAttrName1, nullptr)]] void FunctionCEName1() {}
```

```
$ clang++ -cc1 -fsycl-is-device -fsyntax-only p.cpp
clang++: github.com/intel/llvm/build/tools/clang/include/clang/AST/Attrs.inc:11988: llvm::SmallVector<std::pair<std::__cxx11::basic_strin
g<char>, std::__cxx11::basic_string<char> >, 4> clang::SYCLAddIRAttributesFunctionAttr::getFilteredAttributeNameValuePairs(const std::optional<llvm::SmallSet<
llvm::StringRef, 4> >&, const clang::ASTContext&) const: Assertion `NameStr && "Attribute name is not a valid string."' failed.
PLEASE submit a bug report to https://github.com/intel/llvm/issues and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: ./bin/clang++ -cc1 -fsycl-is-device -fsyntax-only p.cpp
1.      <eof> parser at end of file
2.      p.cpp:3:123: parsing function body 'FunctionCEName1'
```

Because the test also contains lines that cause compilation errors, we exit early from parsing/handling some of the things and therefore we don't see this assert. With the attribute arguments analysis performed earlier, we always go through that path and any "early exits" due to errors in the code do not help us avoid this issue.